web/src/components/netflow-tab.tsx

```diff
@@ -28,18 +39,44 @@ type HPAProps = K8sResourceCommon & {
 
 export const NetflowTab: React.FC<PageComponentProps> = ({ obj }) => {
   const { t } = useTranslation('plugin__netobserv-plugin');
+  const [loading, setLoading] = React.useState(true);
+  const initState = React.useRef<Array<'initDone' | 'configLoading' | 'configLoaded' | 'forcedFiltersLoaded'>>([]);
+  const [config, setConfig] = React.useState<Config>(defaultConfig);
   const [forcedFilters, setForcedFilters] = React.useState<Filters | null>(null);
-  const previousObj = usePrevious(obj);
+  const previous = usePrevious({ obj, state: initState.current });
+
+  React.useEffect(() => {
```

Comment:
the same logic here is done in netflow-traffic component .. would it make sense to move that in the parent, and pass it down to its children? (unless it's not as simple as that)

Reply:
The init state of the tab is not the same as the init state of netflow traffic here.

I've replicated the `getConfig` part that could be shared in a common parent but it will need a bit of refactoring and to find a way to discriminate tabs from page.
Currently we are pointing different components using `"$codeRef": "netflowParent.default"` / `"$codeRef": "netflowTab.default"`

If you want we can try that in a followup

Reply:
no no that's fine, I thought it was really the same thing